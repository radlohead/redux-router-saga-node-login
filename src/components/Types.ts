import { FormSubmitHandler } from "redux-form";
import { FormEvent } from "react";

export interface IJoinSubmitType {
    name?: string;
    id?: string;
    password?: string;
}

export interface IJoinFormType {
    handleSubmit(e: FormSubmitHandler<{}, {}, string>|FormEvent<HTMLFormElement>): void;
}

export interface ILoginSubmitType {
    id?: string;
    password?: string;
}

export interface ILoginFormType {
    handleSubmit(e: FormSubmitHandler<{}, {}, string>|FormEvent<HTMLFormElement>): void;
}