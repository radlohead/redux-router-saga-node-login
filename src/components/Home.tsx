import * as React from 'react';
import { Login } from './index';

const Home = () => {
    return (
        <>
            <Login />
        </>
    )
}

export default Home;