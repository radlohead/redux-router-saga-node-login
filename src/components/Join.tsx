import * as React from 'react';
import { JoinForm } from './index';
import { IJoinSubmitType } from './Types';
import { getJoinFetchApi } from '../sagas';

class Join extends React.Component {
    handleSubmit(values: IJoinSubmitType): void {
        getJoinFetchApi(values);
    }

    render(): JSX.Element {
        return (
            <>
                <JoinForm onSubmit={this.handleSubmit.bind(this)} />
            </>
        )
    }
}

export default Join;