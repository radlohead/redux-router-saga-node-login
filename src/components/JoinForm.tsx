import * as React from "react";
import { Field, reduxForm } from 'redux-form';
import { IJoinFormType } from './Types';

const JoinForm = (props: IJoinFormType) => {
    const { handleSubmit } = props;
    return (
        <form onSubmit={handleSubmit}>
            <div>
                <label>이름</label>
                <div>
                    <Field
                        name="name"
                        component="input"
                        type="text"
                        placeholder="이름을 입력해 주세요"
                    />
                </div>
                <div>
                    <label>아이디</label>
                    <div>
                        <Field
                            name="id"
                            component="input"
                            type="text"
                            placeholder="아이디를 입력해 주세요"
                        />
                    </div>
                </div>
                <div>
                    <label>패스워드</label>
                    <div>
                        <Field
                            name="password"
                            component="input"
                            type="password"
                            placeholder="비밀번호를 입력해 주세요"
                        />
                    </div>
                </div>
            </div>
            <div>
                <button type="submit">입력 완료</button>
            </div>
        </form>
    )
}

export default reduxForm({
    form: 'joinForm'
})(JoinForm);