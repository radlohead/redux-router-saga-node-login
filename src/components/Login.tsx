import * as React from 'react';
import { LoginForm } from './index';
import { ILoginSubmitType } from './Types';
import { getLoginFetchApi } from '../sagas';
import axios from 'axios';
import * as actions from '../actions';

class Login extends React.Component {
    handleSubmit(values: ILoginSubmitType): void {
        getLoginFetchApi(values);

        const postLoginFetchApi = async (values?: any) => {
            const response = await axios.post(`${actions.BASE_SERVER_URL}/api/login`, values);
            console.log('handleSubmit', response.data, values);
        }
        postLoginFetchApi(values);
    }

    render(): JSX.Element {
        return (
            <>
                <LoginForm onSubmit={this.handleSubmit} />
            </>
        )
    }
}

export default Login;