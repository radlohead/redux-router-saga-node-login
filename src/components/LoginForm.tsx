import * as React from "react";
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import { ILoginFormType } from './Types';

class LoginForm extends React.Component<ILoginFormType> {
    public render(): JSX.Element {
        const { handleSubmit }: ILoginFormType = this.props;
        return (
            <form onSubmit={handleSubmit}>
                <div>
                    <div>
                        <label>아이디</label>
                        <div>
                            <Field
                                name="id"
                                component="input"
                                type="text"
                                placeholder="아이디를 입력해 주세요"
                            />
                        </div>
                    </div>
                    <div>
                        <label>패스워드</label>
                        <div>
                            <Field
                                name="password"
                                component="input"
                                type="password"
                                placeholder="비밀번호를 입력해 주세요"
                            />
                        </div>
                    </div>
                </div>
                <div>
                    <button type="submit">로그인</button>
                </div>
            </form>
        )
    }
}

const loginForm = reduxForm({
    form: 'loginForm'
})(LoginForm as any);

export default connect(null)(loginForm);