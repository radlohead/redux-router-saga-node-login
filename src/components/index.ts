export { default as Menu } from './Menu';
export { default as Home } from './Home';
export { default as Join } from './Join';
export { default as JoinForm } from './JoinForm';
export { default as Login } from './Login';
export { default as LoginForm } from './LoginForm';