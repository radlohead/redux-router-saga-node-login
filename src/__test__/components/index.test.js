import * as index from '../../components';

describe('components index', () => {
    let { Menu, Home, Join, JoinForm, Login, LoginForm } = index;
    
    it('Menu', () => expect(Menu).toMatchSnapshot());
    it('Home', () => expect(Home).toMatchSnapshot());
    it('Join', () => expect(Join).toMatchSnapshot());
    it('JoinForm', () => expect(JoinForm).toMatchSnapshot());
    it('Login', () => expect(Login).toMatchSnapshot());
    it('LoginForm', () => expect(LoginForm).toMatchSnapshot());
});