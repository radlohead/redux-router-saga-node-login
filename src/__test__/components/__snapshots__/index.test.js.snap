// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components index Home 1`] = `[Function]`;

exports[`components index Join 1`] = `[Function]`;

exports[`components index JoinForm 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "destroyOnUnmount": true,
    "enableReinitialize": false,
    "forceUnregisterOnUnmount": false,
    "form": "joinForm",
    "getFormState": [Function],
    "keepDirtyOnReinitialize": false,
    "persistentSubmitErrors": false,
    "pure": true,
    "shouldAsyncValidate": [Function],
    "shouldError": [Function],
    "shouldValidate": [Function],
    "shouldWarn": [Function],
    "touchOnBlur": true,
    "touchOnChange": false,
    "updateUnregisteredFields": false,
  },
  "displayName": "ReduxForm",
  "render": [Function],
}
`;

exports[`components index Login 1`] = `[Function]`;

exports[`components index LoginForm 1`] = `[Function]`;

exports[`components index Menu 1`] = `[Function]`;
